### ISSUE TITLE
[expo-dev-menu] is broken for landscape orientation

### ISSUE THREAD

**Issue by todorone** [expo/expo#18005]
### Summary

`expo-dev-menu` is broken for projects that have `"orientation": "landscape"` in `app.json`.
`expo-dev-menu` is hardcoded for `portrait` orientation and is not usable in landscape(button on initial screen is not pressable)

Affects both Expo Go and app with `expo-dev-client`

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

both bare and managed

### What platform(s) does this occur on?

Android, iOS

### SDK Version (managed workflow only)

45

### Environment

  expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.3.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
      Watchman: 4.9.0 - /opt/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/lib/ruby/gems/3.0.0/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
      Android SDK:
        Android NDK: 21.3.6528147
    IDEs:
      Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~45.0.0 => 45.0.5
      react: 17.0.2 => 17.0.2
      react-dom: 17.0.2 => 17.0.2
      react-native: 0.68.2 => 0.68.2
      react-native-web: 0.17.7 => 0.17.7
    npmGlobalPackages:
      eas-cli: 0.52.0
      expo-cli: 5.4.12
    Expo Workflow: bare

### Reproducible demo

Fresh project created with expo-cli that has `"orientation": "landscape"` in `app.json`

**Comment by todorone** [comment]
Error callstack on iOS:
```
2022-06-28 12:07:51.777267+0300 boom[8143:2232371] DevLauncher tries to handle uncaught exception: Supported orientations has no common orientation with the application, and [EXDevMenu.DevMenuViewController shouldAutorotate] is returning YES
2022-06-28 12:07:51.871209+0300 boom[8143:2232371] Stack Trace: (
    "0   CoreFoundation                      0x000000020cb37198 <redacted> + 252",
    "1   libobjc.A.dylib                     0x000000020bd0f9f8 objc_exception_throw + 56",
    "2   CoreFoundation                      0x000000020ca413a0 <redacted> + 0",
    "3   UIKitCore                           0x0000000238f8236c <redacted> + 920",
    "4   UIKitCore                           0x0000000238f72968 <redacted> + 96",
    "5   UIKitCore                           0x0000000239545b20 <redacted> + 120",
    "6   boom                              0x0000000100e738b0 $s9EXDevMenu03DevB6WindowC7managerAcA0cB7ManagerC_tcfc + 412",
    "7   boom                              0x0000000100e73704 $s9EXDevMenu03DevB6WindowC7managerAcA0cB7ManagerC_tcfC + 40",
    "8   boom                              0x0000000100e63e68 $s9EXDevMenu03DevB7ManagerCACycfc + 628",
    "9   boom                              0x0000000100e6409c $s9EXDevMenu03DevB7ManagerCACycfcTo + 20",
    "10  boom                       
2022-06-28 12:07:51.912987+0300 boom[8143:2232371] *** Terminating app due to uncaught exception 'UIApplicationInvalidInterfaceOrientation', reason: 'Supported orientations has no common orientation with the application, and [EXDevMenu.DevMenuViewController shouldAutorotate] is returning YES'
*** First throw call stack:
(0x20cb37180 0x20bd0f9f8 0x20ca413a0 0x238f8236c 0x238f72968 0x239545b20 0x100e738b0 0x100e73704 0x100e63e68 0x100e6409c 0x100e622ac 0x100e6227c 0x1043e8c78 0x1043eac84 0x23a8f6e40 0x100e62300 0x100e6231c 0x100e62360 0x100d4a0b8 0x100d463b4 0x100d46164 0x100d462f8 0x100d64904 0x100d64d08 0x10084ab08 0x100840934 0x100840780 0x100840274 0x100840844 0x23a795664 0x1008405cc 0x100840a20 0x1007c0200 0x2395070f0 0x239508854 0x23950dfe0 0x238dd12a4 0x238dd983c 0x238dd0f28 0x238dd1818 0x238dcfb64 0x238dcf82c 0x238dd436c 0x238dd5150 0x238dd4224 0x238dd8f24 0x23950c5e8 0x239108e04 0x20f4b49fc 0x20f4be40c 0x20f4bdc14 0x1043e8c78 0x1043ec840 0x20f4ef040 0x20f4eecdc 0x20f4ef294 0x20cac8f1c 0x20cac8e9c terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'UIApplicationInvalidInterfaceOrientation', reason: 'Supported orientations has no common orientation with the application, and [EXDevMenu.DevMenuViewController shouldAutorotate] is returning YES'
0x20cac8784 0x20cac36c0 0x20cac2fb4 0x20ecc579c 0x23950fc38 0x1007c0acc 0x20c5868e0)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

**Comment by chris24elias** [comment]
landscape broken for me as well

**Comment by hirbod** [comment]
This is how the dev menu looks like in my landscape-only iPad app :D

<img width="1173" alt="Bildschirmfoto 2022-07-19 um 12 18 58" src="https://user-images.githubusercontent.com/504909/179727918-3b01045c-9fed-4ab2-9f83-c4cc5123dbc7.png">

**Comment by keith-kurak** [comment]
This is fixed in https://github.com/expo/expo/pull/18509 and is available in SDK 46/ `expo-dev-client` 1.2.1
